src/csharp/Microsoft.Spark.E2ETest/IpcTests/BroadcastTests.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Spark.Sql;
+using static Microsoft.Spark.Sql.Functions;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Microsoft.Spark.E2ETest.IpcTests
+{
+
+    [Serializable]
+    public class TestBroadcastVariable
+    {
+        public int IntValue { get; set; }
+        public string StringValue { get; set; }
+
+        public TestBroadcastVariable(int intVal, string stringVal)
+        {
+            IntValue = intVal;
+            StringValue = stringVal;
+        }
+    }
+
+    [Collection("Spark E2E Tests")]
+    public class BroadcastTests
+    {
+        private readonly SparkSession _spark;
+        private readonly DataFrame _df;
+
+        public BroadcastTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+            var data = new List<string>(new string[] { "hello ", "world " });
+            _df = _spark.CreateDataFrame(data);
+        }
+
+        /// <summary>
+        /// Test Broadcast support by using a single broadcast variable in a UDF.
```

Comment:
Don't you need to clean up broadcast registry after each test or after this suite?

Reply:
Yes, thanks for pointing that out! I figured it would get cleaned up after the application finished but it is better practice to clean after each test. Or would you prefer to clean up after the suit?